Services/SwissBitcoinPayService.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Cache;
+using System.Security.Cryptography;
+using System.Text;
+using System.Web;
+using Newtonsoft.Json;
+using Payments.SwissBitcoinPay;
+using Payments.SwissBitcoinPay.Models;
+using Org.BouncyCastle.Utilities;
+using System.Net.Http;
+
+namespace Payments.SwissBitcoinPay.Services
+{
+    public class SwissBitcoinPayService
+    {
+
+        public static bool CheckSecretKey(string key, string message, string signature)
+        {
+            var msgBytes = HMACSHA256.HashData(Encoding.UTF8.GetBytes(key), Encoding.UTF8.GetBytes(message));
+            string hashString = string.Empty;
+            foreach (byte x in msgBytes)
+            {
+                hashString += String.Format("{0:x2}", x);
+            }
+            return (hashString == signature);
+        }
+
+        public string CreateInvoice(SwissBitcoinPaySettings settings, PaymentDataModel paymentData)
+        {
+
+            try
+            {
+                var invoice = new SwissBitcoinPayInvoiceModel()
+				{
+                    title = paymentData.Description,
+                    description = $"{paymentData.BuyerName} | Order : {paymentData.OrderID} | Store : {paymentData.StoreID.ToString()}",

```

Comment:
I think to record the StoreID is not necessary.
```suggestion
                    description = $"{paymentData.BuyerName} | Order : {paymentData.OrderID}",

```

Reply:
No, the Store ID is mandatory in the WebHook for GrandNode

Reply:
Then I suggest you pass this value in the `extra` field (for ex. as `storeId`), so you can get it back in your webhook function via `jsonData.extra.storeId`

The storeId shouldn't be explicitly displayed to the user, it's not valuable.